src/chunk.c

```diff
@@ -180,7 +188,7 @@ ChunkResult Uncompressed_ChunkIteratorGetPrev(ChunkIter_t *iterator, Sample *sam
 void Uncompressed_FreeChunkIterator(ChunkIter_t *iterator) {
     ChunkIterator *iter = (ChunkIterator *)iterator;
     if (iter->options & CHUNK_ITER_OP_FREE_CHUNK) {
-        free(iter->chunk);
+        Uncompressed_FreeChunk(iter->chunk);
```

Comment:
This is the line that actually fixes the leak. The rest along with #515 is unrelated to the leak